modules/rosa-cluster-logging-collector-log-forward-sts-cloudwatch.adoc

```diff
@@ -0,0 +1,195 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-external.adoc
+
+:_content-type: PROCEDURE
+
+[id="cluster-logging-collector-log-forward-sts-cloudwatch_{context}"]
+= Forwarding logs to Amazon CloudWatch from STS enabled clusters
+
+For clusters with AWS Security Token Service (STS) enabled, create the AWS IAM roles and policies that will allow the log forwarding, and a `ClusterLogForwarder` custom resource (CR) with an output for CloudWatch.
+
+[NOTE]
+====
+This feature is not supported by the vector collector.
+====
+
+.Prerequisites
+
+* {logging-title-uc}: 5.5 and later
+
+.Procedure
+. Prepare the AWS account:
+.. Create an IAM policy JSON file with the following content:
++
+[source,json]
+----
+{
+"Version": "2012-10-17",
+"Statement": [
+    {
+    "Effect": "Allow",
+    "Action": [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+      "logs:PutRetentionPolicy"
+    ],
+    "Resource": "arn:aws:logs:*:*:*"
+    }
+  ]
+}
+----
++
+.. Create an IAM trust JSON file with the following content:
++
+[source,json]
+----
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::<you_aws_account_id>:oidc-provider/<openshift_oidc_provider>" <1>
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "<openshift_oidc_provider>:sub": "system:serviceaccount:openshift-logging:logcollector" <2>
+        }
+      }
+    }
+  ]
+}
+----
++
+--
+<1> Specify your AWS account ID and the OpenShift OIDC provider endpoint. Obtain the endpoint by running the the following command:
++
+[source,terminal]
+----
+$ rosa describe cluster \
+  -c $(oc get clusterversion -o jsonpath='{.items[].spec.clusterID}{"\n"}') \
+  -o yaml | awk '/oidc_endpoint_url/ {print $2}' | cut -d '/' -f 3,4
+----
++
+<2> Specify the OpenShift OIDC endpoint again.
+--
+
+.. Create the IAM role:
++
+[source,terminal]
+----
+$ aws iam create-role
+  --role-name “<your_rosa_cluster_name>-RosaCloudWatch” \
+  --assume-role-policy-document file://<your_trust_file_name>.json \
+  --query Role.Arn \
+  --output text
+----
++
+Save the output. You will use it in the next steps.
++
+.. Create the IAM policy:
++
+[source,terminal]
+----
+$ aws iam create-policy \
+--policy-name "RosaCloudWatch" \
+--policy-document file:///<your_policy_file_name>.json \
+--query Policy.Arn \
+--output text
+----
++
+Save the output. You will use it in the next steps.
+
+.. Attach the IAM policy to the IAM role:
++
+[source,terminal]
+----
+$ aws iam attach-role-policy \
+ --role-name “<your_rosa_cluster_name>-RosaCloudWatch” \
+ --policy-arn <policy_ARN> <1>
+----
++
+<1> Replace `policy_ARN` with the output you saved while creating the policy.
+
+. Create a `Secret` YAML file for the logging operator:
++
+--
+[source,yaml]
```

Comment:
Need to confirm with Anping Li  if this secret is OK (there are two types).  Also need confirmation that the procedure works as expected.

Reply:
@anpingli could you please have a look at this PR and recommend the next steps.

Reply:
/lgtm.  Confirmed this type of secret works.